types/lodash/fp.d.ts

```diff
@@ -1997,7 +1997,7 @@ declare namespace _ {
     type LodashIsNumber = (value: any) => value is number;
     type LodashIsObject = (value: any) => value is object;
     type LodashIsObjectLike = (value: any) => boolean;
-    type LodashIsPlainObject = (value: any) => boolean;
+    type LodashIsPlainObject = (value: any) => value is object;
```

Comment:
The problem is that it's currently impossible to type guard `isPlainObject` correctly. `value is object` is NOT correct in the case where `isPlainObject` returns false for a non-plain object. For example:

```ts
class Foo {...}

let obj: Foo|{};
if (isPlainObject(obj)) {
  // obj is Foo|{} -- ok
} else {
  // obj is never -- wrong!
}
```